lib/bson/decimal128.rb

```diff
@@ -107,11 +107,18 @@ def ==(other)
     #
     # @since 4.2.0
     def initialize(initial, digits = nil)
-      if initial.is_a?(String)
-        set_bits(*Builder::FromString.new(initial).bits)
-      else
-        big_decimal = BigDecimal.new(*[initial, digits].compact)
-        set_bits(*Builder::FromBigDecimal.new(big_decimal).bits)
+      case initial
+        when String
+          set_bits(*Builder::FromString.new(initial).bits)
+        when Integer
+          set_bits(*Builder::FromInteger.new(initial).bits)
+        when Float
+          set_bits(*Builder::FromFloat.new(initial).bits)
```

Comment:
Float can have all sorts of precision related errors, for example try this in IRB:

139.25 + 74.79

We should at least make `digits` optional if not required. (BigDecimal requires it)